src/Tasks/Microsoft.Common.CurrentVersion.targets

```diff
@@ -889,6 +889,7 @@ Copyright (C) Microsoft Corporation. All rights reserved.
     -->
   <PropertyGroup>
     <BuildDependsOn>
+      $(BuildDependsOn);
```

Comment:
My naive thinking was that the options in order from *least* preferred (from my point of view):

1. Do nothing - let the situation as is
2. Consider each of the DependsOn* property separately and decide whether to:
   * Leave untouched (cannot be extended)
   * Allow prepending
   * Allow appending
4. Perform a blanket fix and unify the behavior
5. Perform a blanket fix and introduce prepend and append options (e.g. `DependsOn***Initial`, `DependsOn***Trialing`)

The **1** and **2** feel very unsystematic and source of confusion for users.
The 5 feels too overstretched to me - but I'm very eager to hear the specific concerns with the `BuildDependsOn`

The **4** goes inline with what we already had in targets plus what was previously fixed in 2 different rounds: https://github.com/dotnet/msbuild/pull/4922 and https://github.com/dotnet/msbuild/pull/9704 
This PR only makes it general so we do not have to discover and fix piece by piece. 

Let me know your concerns or alternate suggestions

Reply:
Friendly ping on this @rainersigwald 😉

Reply:
I think we have to think them through individually. Changes like this feel way worse to me than the existing behavior.

Reply:
Ok having looked through the PR more I think the default should be to _append_ the previous value. That's generally more in keeping with what I expect is the primary expectation: "run this new thing right before the X in XDependsOn".

In the old explicit-import model users had flexibility either way though, so I'm kinda wondering whether this is the right thing to do at all.

Reply:
I need your help understanding the concern - by which I do not mean I disagree - I'm just genuinely confused.

With this change:
 * The explicit-import model can still be used and will remain identical to what the behavior was before (including the flexibility of usage of order)
 * The behavior of DependsOn properties will now be applied to all. Using the same mechanism (prepending) as was used for the other DependsOn properties in common targets, that were already authored to be nondestructive, and to all DependsOn properties in common targets, that were fixed in previous 2 approved and merged PRs.


I believe the unification brings clarity and is an improvement, but if the change would feel degrading or dangerous - I'm definitely fine closing it. Concrete cases or reasons would help me understand that better

Reply:
My question is whether prepending meets user expectations. If the user says

```xml
<BuildDependsOn>$(BuildDependsOn);MyCustomThing</BuildDependsOn>
```

in the body of their project, do they expect that `MyCustomThing` will be _the first target that runs in the whole build_? Because that's what this change will do.

Reply:
Understood.

Unfortunately same concerns applies the other way around (`<BuildDependsOn>MyCustomThing;$(BuildDependsOn)</BuildDependsOn>` in user code, then extended by `<BuildDependsOn>CommonTargetEntries;$(BuildDependsOn)</BuildDependsOn>` in common targets doesn't lead to the custom target being the first thing to run). In both cases the previous behaavior was anyway ignored with the implicit import and in both cases it'll work as intended with explicit import.

That let me to keep the behavior in line what was already extendable or fixed before.

How about flipping everything to append (including the preexisting prepends)? It feels as OCD trigger if there'd be unjustified exceptions 😄 - so I'm trying to see if/how we can unify.

Reply:
Yeah I'd say _in general_ append feels more like the general policy: it's more like what you'd get if you hooked the same target with `BeforeTargets`, which is what I'd generally recommend. But I am as usual scared of making a user-visible change like that :-/

Reply:
OK - that is fair - let's go far appending the custom targets in all cases.
@AArnott, @baronfel - would you have any concerns if I adjust the DependsOn* chaining from your fixes (#4922 and #9704) and put the user defined targets to the end of the list (instead to start)?

Reply:
It's hard to predict whether changing target order will break something. If it does, it can usually be corrected by adding another dependency to one of the targets in the list.
I think the most important thing is that there is _some_ way for a user to prepend and append targets to the property's final value, for when ordering matters. Ideally, the built-in targets would be defined on the property in the .props area, so that the project file itself could simply choose whether to prepend or append. But if these properties have always been defined in .targets, then they have the initial say as to whether to append or prepend their own targets to what the project file may have given. If we can arrange code such that the built-in .targets have appended/prepended to these properties before `Directory.Build.targets` is imported then the user can choose to deliberately append/prepend to that at that point.

I can't predict whether rearranging the ordering from my PR will break my use or anyone else's. My guess is it will *probably* be fine. Generally though, my default for property lists is to append by default, and prepend when needed, which is why I wrote the PR as I did.

BTW, probably water under the bridge at this point as the precedent is set, but IMO using properties for this is wrong. In MSBuild, properties are scalars while items are vectors. A target's DependsOn attribute wants a vector, so why aren't we using items for these lists of targets? It also adds the ability to `Remove` items more conveniently than is required for properties. FWIW, when I define my own msbuild targets in my repos, I use msbuild items for this and they work great.

Reply:
Thank you @AArnott and @rainersigwald for the perspective. I moved to the append pattern (including couple pre-existing cases). There will still remain to old-school way of explicit importing sdk imports and then prepending/appending as needed.

It's be great if you could have a second look and see if it feels as an improvement or rather as a risk with questionable benefits - so that we can decide whther to iterate on this or rather just close (no regrets :))

Thank you!